client.go

```diff
@@ -448,8 +446,8 @@ func (c *Client) connect(identifier, serverAddr string) error {
 	c.log.Debug("client has started successfully.")
 	c.redialBackoff.Reset() // we successfully connected, so we can reset the backoff
 
-	c.mu.Lock()
-	if c.startNotify != nil && !c.closed {
+	c.closedMu.RLock()
+	if !c.closed && c.startNotify != nil {
```

Comment:
I can fix that, I left it because this way no closing can interrupt sending notification, @cihangir what do you think?